vault/identity_store_entities.go

```diff
@@ -769,15 +812,20 @@ func (i *IdentityStore) mergeEntity(ctx context.Context, txn *memdb.Txn, toEntit
 		}
 	}
 
+	// Check alias clashes after validating every fromEntity, so that we have a full list of errors
+	if aliasClashError != nil {
+		return aliasClashError, nil
+	}
+
 	isPerfSecondaryOrStandby := i.localNode.ReplicationState().HasState(consts.ReplicationPerformanceSecondary) ||
 		i.localNode.HAState() == consts.PerfStandby
 	var fromEntityGroups []*identity.Group
 
-	toEntityAccessors := make(map[string]struct{})
+	toEntityAccessors := make(map[string]string)
 
 	for _, alias := range toEntity.Aliases {
 		if _, ok := toEntityAccessors[alias.MountAccessor]; !ok {
-			toEntityAccessors[alias.MountAccessor] = struct{}{}
+			toEntityAccessors[alias.MountAccessor] = alias.ID
```

Comment:
This is saving only the last alias in the aliases list that is tied to a mount accessor. I think we would need this to be []string.

Reply:
There shouldn't be multiple aliases in the toEntity with the same mount accessor to begin with, unless something has gone very wrong - I think we're safe

Reply:
Chatted with Roy about this - since we don't actually enforce for sure that there aren't multiple entity aliases sharing a mount accessor, it's best to use slices just to handle the multiple alias case in case it happens.